docs/spec.openapi.yaml

```diff
@@ -0,0 +1,425 @@
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+  /files/{fileId}/edit:
+    put:
+      tags:
+        - Files
+      description: Edit an existing file
+      parameters:
+        - in: path
+          name: fileId
+          required: true
+          schema:
+            type: string
+        - in: query
+          name: newContent
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: File successfully edited
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '401':
+          description: Not authorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+  /files/{fileId}/delete:
+    delete:
+      tags:
+        - Files
+      description: Deletes an existing file
+      parameters:
+        - in: path
+          name: fileId
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: File successfully deleted
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '401':
+          description: Not authorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description:  Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+  /files/{fileId}/move:
+    patch:
+      tags:
+        - Files
+      description: Move an existing file to a different location
+      parameters:
+        - in: path
+          name: fileId
+          required: true
+          schema:
+            type: string
+        - in: query
+          name: newLocation
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: File moved successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '401':
+          description: Not authorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+
+  /logout:
```

Comment:
Since the Authentication Micro-service is not sending the JWT as a cookie there is no reason to have an endpoint for the logout. The clients can just remove the JWT from the `Session Storage` or any other client-side place.